src/qutip_qip/device/modelprocessor.py

```diff
@@ -243,6 +243,33 @@ def load_circuit(self, qc, schedule_mode="ASAP", compiler=None):
         self.set_tlist(tlist)
         return tlist, coeffs
 
+    def _get_density_matrix(self, result):
+        """
+        Convert a processor result to a density matrix.
+        """
+        density_matrix = None
+        if len(result.states):
+            density_matrix = result.states[-1]
+            # if answer is in ket form, convert
+            # to density matrix with outer product
+            if density_matrix.type == "ket":
+                density_matrix = density_matrix * density_matrix.dag()
+        return density_matrix
```

Comment:
Why do we need this? If there are no collapse operators. Shouldn't we return just a ket?